File: Scala_example/PrimeBenchmark.scala
object PrimeBenchmark extends App {
    def sieveOfEratosthenes(n: Int): Array[Boolean] = {
        val primes = Array.fill(n + 1)(true)
        primes(0) = false
        primes(1) = false
        
        for (i <- 2 to math.sqrt(n).toInt if primes(i)) {
            for (j <- i * i to n by i) {
                primes(j) = false
            }
        }
        primes
    }

    val n = 1000000  // Adjust as needed
    val primes = sieveOfEratosthenes(n)
    // Print prime numbers, if needed
    primes.zipWithIndex.filter(_._1).map(_._2).foreach(println)
}

File: Java_example/PrimeBenchmark.java
import java.util.Arrays;

public class PrimeBenchmark {
    public static boolean[] sieveOfEratosthenes(int n) {
        boolean[] primes = new boolean[n + 1];
        Arrays.fill(primes, true);
        primes[0] = primes[1] = false;
        
        for (int i = 2; i <= Math.sqrt(n); i++) {
            if (primes[i]) {
                for (int j = i * i; j <= n; j += i) {
                    primes[j] = false;
                }
            }
        }
        return primes;
    }

    public static void main(String[] args) {
        int n = 1000000; // Adjust as needed
        boolean[] primes = sieveOfEratosthenes(n);
        //Print prime numbers, if needed
        
        for (int i = 2; i <= n; i++) {
            if (primes[i]) {
                System.out.println(i);
            }
        }
        
    }
}
